Connect PostrgreSQL

NazariiTkach opened the pull request:
2 - Connect PostrgreSQL
3 - Create Products controller (Backend)
4 - User Porducts api in the forontend

NazariiTkach committed:
instal expres server

localhost:3000

NazariiTkach committed:
Merge pull request #1 from NazariiTkach/expres-server

install expres server

NazariiTkach committed:
Connect PostrgreSQL

2 - Connect PostrgreSQL
3 - Create Products controller (Backend)
4 - User Porducts api in the forontend

NazariiTkach committed:
corrections based on comments

remove LiveReload

Halynsky committed:
Merge branch 'main' of https://github.com/NazariiTkach/training-project into Connect-PostrgreSQL(with-Docker-or-DockerCompose)

# Conflicts:
#	package-lock.json
#	package.json
#	public/index.js
#	readme

NazariiTkach committed:
remove node modules

NazariiTkach committed:
fixed gitignore

NazariiTkach committed:
add node modules to gitignore

Halynsky commented on backend/controllers/productsController.js:
```diff
@@ -0,0 +1,14 @@
+
+async function getAllProducts(req, res) {
+    try {
+      const result = await pool.query('SELECT name, price, image_url FROM products');
```
Why not select *

NazariiTkach committed:
Product creation and deletion

Add plus button for Product creation. Add modal dialog for creation. Add delete button.  Show updated products list after createn.
Сode optimization

Halynsky commented on backend/db-init.js:
```diff
@@ -0,0 +1,33 @@
+  } finally {
+    await pool.end();
+  }
+}
```
Its not good that DDL and DML are in same function
better do 2 functions
fist for initializing the DB structure and the second one for the data

Halynsky commented:
If you are making an UI its nice to show a screenshot in the PR

Halynsky reviewed (changes requested):
Check comments

NazariiTkach commented:
Added buttons for creating and deleting a product
"+" - button to create a product
"x"- button to remove the product

![Buttons](https://github.com/user-attachments/assets/cff49557-39bb-4e25-a6d1-3cc7c3f7e950)

------------------------------------------------------------------------------
Added modal window for creating a product
The modal opens when the "+" button is clicked
It includes fields for name, price
After submission, the product is added to the table

![Modal window](https://github.com/user-attachments/assets/3a571828-b276-40bf-a57b-8150ed59942c)

NazariiTkach committed:
change db-init.js

created: function initSchema and function seedData

NazariiTkach committed:
Product sorting added

changed the design of the modal window for adding a new product, the buttons for adding and deleting a product. added a dialog box for confirming the deletion of a product
